chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/AssaultException.java

```diff
@@ -1,100 +1,132 @@
-        instance = constructor.newInstance();
-      } else {
-        Constructor<? extends Throwable> constructor =
-            exceptionClass.getConstructor(this.getExceptionArgumentTypes().toArray(new Class[0]));
-        instance =
-            constructor.newInstance(this.getExceptionArgumentValues().toArray(new Object[0]));
-      }
-    } catch (ReflectiveOperationException e) {
-      Logger.warn(
-          "Cannot instantiate the class for provided type: {}. Fallback: Throw RuntimeException",
-          type);
-      instance = new RuntimeException("Chaos Monkey - RuntimeException");
+      ThrowableCreator creator = getCreator();
+      instance = creator.create(getExceptionArgumentValues());
+    } catch (ReflectiveOperationException | ClassCastException | JsonProcessingException e) {
+      Logger.warn("Failed to create custom exception. Fallback: Throw RuntimeException");
+      instance = new RuntimeException("Chaos Monkey - RuntimeException (Fallback)", e);
     }
-
-    throw instance;
+    return instance;
   }
 
   @JsonIgnore
-  public Class<? extends Throwable> getExceptionClass() throws ClassNotFoundException {
-    if (type == null) {
-      // use Chaos Monkey default Runtime Exception
-      type = "java.lang.RuntimeException";
-      ExceptionArgument exceptionArgument = new ExceptionArgument();
-      exceptionArgument.setClassName("java.lang.String");
-      exceptionArgument.setValue("Chaos Monkey - RuntimeException");
-      arguments = Collections.singletonList(exceptionArgument);
+  public ThrowableCreator getCreator() throws ReflectiveOperationException {
+    Class<?> exceptionClass = getExceptionClass();
+    Class<?>[] argumentTypes = getExceptionArgumentTypes().toArray(new Class[0]);
+    if ("<init>".equals(method)) {
+      return new ThrowableConstructor(
+          exceptionClass.asSubclass(Throwable.class).getConstructor(argumentTypes));
+    } else {
+      return new ThrowableStaticInitializer(exceptionClass.getMethod(method, argumentTypes));
     }
-    return Class.forName(type).asSubclass(Throwable.class);
   }
 
   @JsonIgnore
-  public List<Class> getExceptionArgumentTypes() {
-    return arguments.stream().map(ExceptionArgument::getClassType).collect(Collectors.toList());
+  public Class<?> getExceptionClass() throws ClassNotFoundException {
+    return Class.forName(type);
   }
 
   @JsonIgnore
-  private List<Object> getExceptionArgumentValues() {
-    return arguments.stream().map(ExceptionArgument::getValue).collect(Collectors.toList());
+  public List<Class<?>> getExceptionArgumentTypes() throws ClassNotFoundException {
+    List<Class<?>> list = new ArrayList<>();
+    for (ExceptionArgument argument : arguments) {
+      list.add(argument.getClassType());
+    }
+    return list;
+  }
+
+  @JsonIgnore
+  private List<Object> getExceptionArgumentValues()
+      throws ClassNotFoundException, JsonProcessingException {
+    List<Object> list = new ArrayList<>();
```

Comment:
lets rename this variable into something like exceptionArguments or something like this (or just values)